src/blocktree.erl

```diff
@@ -13,8 +13,21 @@
 
 -include("potato_records.hrl").
 
-
-
+-type tx() :: map().
+-type txarray() :: array:array(tx()).
+-type txlist() :: [tx()].
+-type playertxmap() :: map().
+-type addresult() :: ignore_duplicate | added | ignore_nonce_too_high.
+-type treedata() :: #tree_data{}.
+-type playerid() :: integer().
+-type nonce() :: integer().
+-type blockid() :: integer() | undefined.
```

Comment:
should this be bytes D:
its bytes for pop protocol, but integers for testing DD:
i'll change it later DDD: